### PR TITLE
Add com.marketo/event/jsonschema/1-0-0

### PR DESCRIPTION
**Pull request by misterpig**
JSON schema for marketo event with all possible fields. Test payloads sent through unpopulated fields as empty strings not nulls. Two fields with format "date". Many fields couldn't be populated so the string length is long (255 and 512) for the vast majority of fields. Marketo only has the one webhook so 3 custom fields are included ("name", "description, "step") to further differentiate the event.

### CODE REVIEW

File: schemas/com.marketo/event/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,601 @@
...
+        },
+        "do_not_call_reason": {
+          "type": "string",
+          "maxLength": 512
+        },
+        "do_not_call": {
+          "type": "boolean"
+        },
+        "email_address": {
+          "type": "string",
```
Comment (chuwy) [inline schemas/com.marketo/event/jsonschema/1-0-0]: There's `email` JSON Schema format.

**Comment by alexanderdean** [comment]
Is there some associated documentation / a PR on how to use this?

**Comment by BenFradet** [comment]
@alexanderdean do you mean snowplow/snowplow#3677 ?

**Comment by alexanderdean** [comment]
Ace! Haven't got to that one in my email yet 🐌

**Comment by snowplowcla** [comment]
@misterpig has signed the Software Grant and Corporate Contributor License Agreement

**Comment by miike** [comment]
@alexanderdean We'll be putting together some documentation on how to get this setup in Marketo this week. The webhook integration is a little odd in that there's only one event that's ever sent (with a customisable payload) but that event can be triggered in multiple steps within a Marketo flow.

This schema represents a superset of these fields with the intention that a customer would likely customise this schema and host in their local Iglu repository rather than relying on this superschema in Iglu Central.

**Comment by miike** [comment]
>I just wanted to remind that you can enforce this schema in adaptor. In other words, this schema should validate not input of adaptor, but output. So if you think this is necessary (I think empty strings instead of null isn't a very good idea) we can make this schema much more stricter.

@chuwy - What would be your advice on this one? Some fields may be sent through correctly as empty strings (which should be stored accordingly) whereas there may also be instances where an empty string should be NULL but is set to the webhook as an empty string. Unfortunately the Marketo [documentation](http://developers.marketo.com/rest-api/lead-database/fields/list-of-standard-fields/) doesn't provide information about which fields are nullable.

**Comment by chuwy** [comment]
@miike if empty strings are valid values and we have no way to distinguish empty-string-as-nulls and valid-empty-strings - I think @misterpig's approach is the correct one.

**Comment by alexanderdean** [comment]
@chuwy what is missing for your green tick?

**Comment by alexanderdean** [comment]
Scheduling!

**Pull request closed without merging**